flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/table/JdbcDynamicTableFactory.java

```diff
@@ -109,6 +109,11 @@
 		.withDescription("gives the reader a hint as to the number of rows that should be fetched, from" +
 			" the database when reading per round trip. If the value specified is zero, then the hint is ignored. The" +
 			" default value is zero.");
+	private static final ConfigOption<Boolean> SCAN_AUTO_COMMIT = ConfigOptions
+		.key("scan.auto-commit")
+		.booleanType()
+		.noDefaultValue()
```

Comment:
I think auto commit mode is not a three-valued (`true`, `false`, `null`), it should be either enabled or disabled. Besides, it should be enabled by default, because this is the default behavior of JDBC. 

So what about changing the default value to `true` and also update the member virables from `Boolean` to primitive `boolean`?

Reply:
I had let it be optional so that it could use the JDBC default. But, per the JDBC spec the default is true, so your suggestion makes sense. I'll make that change.

In the email thread, you had made a comment about setting auto commit to false by default within `JdbcRowDataInputFormat.Builder` in the 1.11 branch so that it would do the right thing for scans in postgres. However, that would conflict with setting the default to be true here. If someone expected that behavior in a 1.11 patch and then we add this, the behavior would revert back if they don't specify this option explicitly.

In general, I think that leaving auto commit enabled is the right thing to do. However, this is a config option specifically for doing scans, so I could see an argument for letting the default be false in this case. But, I can also see a good argument for making it match the JDBC driver default. I would suggest whatever is decided that it should be consistent, so if we set the default to be true here we probably shouldn't change `JdbcRowDataInputFormat.Builder`.

Let me know which direction you'd like to go with this.

Reply:
In the mailing list, I had made a mistake that I thought the `autoCommit` is false by default in `JdbcRowDataInputFormat.Builder` (but it is null by default), so I said it should be true by default. As for now, I think we don't need to add a patch to branch 1.11 because this is not a bug, but is a new feature.

I think it would be more reasonable to make auto commit default true everywhere.